src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -246,11 +245,16 @@ public Image<TPixel> Decode<TPixel>(BufferedReadStream stream, CancellationToken
                                 currentFrameControl.Value,
                                 cancellationToken);
 
-                            previousFrame = currentFrame;
-                            previousFrameControl = currentFrameControl;
+                            // if current frame dispose is restore to previous, then from future frame's perspective, it never happened
+                            if (currentFrameControl.Value.DisposeOperation != PngDisposalMethod.RestoreToPrevious)
```

Comment:
Are you sure about this? We need to ensure we create the entire frame by drawing the current frame over the previous.

Can you please check this against the gif decoder?

Reply:
This is what the GIF decoder does too. When the disposal method is RestoreToPrevious, it sets previousFrame to the value it already had (GifDecoderCore lines 522 and 621). Otherwise, it sets previousFrame to the current frame, updating it for the next one (line 625).